derp/derp_server.go

```diff
@@ -795,6 +824,49 @@ func (c *sclient) run(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
+		c.cleanFlows()
+	}
+}
+
+// flowsTimerFunc tells all the sclients to scan for inactive flows
+// and free them.
+func (s *Server) flowsTimerFunc() {
```

Comment:
It would be nice to be able to have this shutdown when the derp server shuts down so that integration/e2e tests don't leak a periodic task.

The shortest path may be to check s.closed before releasing the lock.

Reply:
Good point. I read a lot of documentation on the difficulty of using timer.Stop() correctly and I think the new version does it properly, using s.mu to synchronize the restart/stop.